src/sap.f/src/sap/f/DynamicPage.js

```diff
@@ -699,13 +699,23 @@ sap.ui.define([
 		bUseAnimations = sAnimationMode !== Configuration.AnimationMode.none && sAnimationMode !== Configuration.AnimationMode.minimal;
 
 		if (exists(this.$contentFitContainer)) {
-			this.$contentFitContainer.toggleClass("sapFDynamicPageContentFitContainerFooterVisible", bShow);
+			if (this.getDomRef()) {
+				var oContentFitContainer = this.getDomRef().querySelector("[id$='contentFitContainer']");
+				if (oContentFitContainer) {
+					oContentFitContainer.classList.toggle("sapFDynamicPageContentFitContainerFooterVisible", bShow);
+				}
+			}
 		}
 
 		if (bUseAnimations) {
 			this._toggleFooterAnimation(bShow, oFooter);
 		} else {
-			this.$footerWrapper.toggleClass("sapUiHidden", !bShow);
+			if (this.getDomRef()) {
+				var oFooterWrapper = this.getDomRef().querySelector("[id$='footerWrapper']");
+				if (oFooterWrapper) {
+					oFooterWrapper.classList.toggle("sapUiHidden", !bShow);
+				}
+			}
 		}
```

Comment:
If jQuery doesn't find any element, calling `.toggleClass` won't throw an error, it won't break the app. The same doesn't apply for vanilla DOM APIs. If `domRef` is null `.querySelector` will throw an error. If `.querySelector` result is null, `.classList.toggle` will throw an error. That's why all the validations.

An utility function could be created to make it easier to use. Something like:
`
function toggleClass(contextDomRef, query, cssClass, condition) {
  if (!contextDomRef || !query || !cssClass) {
    return;
  }
  const selectedElement = contextDomRef.querySelector(query);
  if (!selectedElement) {
    return;
  }
  if (typeof condition === 'boolean') {
    selectedElement.classList.toggle(cssClass, condition);
  } else {
    selectedElement.classList.toggle(cssClass);
  }
}
`

Usage would be like:
`
toggleClass(document.body, 'div', 'someCssClass');
toggleClass(this.getDomRef(), '[id$="some-id-suffix"]', 'someCssClass', 1 > 0);
`

Reply:
That was my point, exactly. I'm in favour of replacing or accompanying our `Element.prototype.$` with a more lightweight substitute, based on vanilla JS. The `classList` mutations are only one example where this might make sense.

Some months ago, we had made experiments with such a substitute. Roughly 95% of the immediate calls to the object returned by `Element.prototype.$` could be easily substituted with vanilla JS. The remaining 5%, however, make a rollout difficult. Replacing `Element.prototype.$` most likely would cause compatibility issues, so accompanying it with a similar API, limited to what can be backed by vanilla JS, makes the most sense to me.

We've however postponed activities reg. such a substitute until we have a plan to remove jQuery completely. Other areas where we use it, namely event handling, are more tricky to replace (as long as compatibility limits us).

These considerations don't prevent the removal of existing jQuery usages, for sure. We also have done it here and there. It just has to be decided case-by-case.

Reply:
I think the main issue would be when `Element.prototype.$` doesn't find anything. Today it always returns a jQuery object and one can call any jQuery method from it, even when `this.getDomRef()` returns null.

With vanilla js, it could cause problems if `Element.prototype.$` doesn't return a jQuery object and returns null. One'd always need to check the returned object before executing any method. In my PR, I've added checks for that reason. Sometimes `this.getDomRef()` or `querySelector` would return null and cause issues.